### PR TITLE
NSFS | NC | Atomic creation of buckets and accounts config files (bucketspace FS and CLI)

### PR DESCRIPTION
**Pull request by romayalon**
### Explain the changes
1. 

### Issues: Fixed #xxx / Gap #xxx
1. https://issues.redhat.com/browse/MCGI-200

### Testing Instructions:
1. 

- [ ] Doc added/updated
- [ ] Tests added

### CODE REVIEW

File: src/cmd/manage_nsfs.js [hunk 1]
```diff
@@ -168,18 +178,26 @@ async function update_bucket_object(config_root, target) {
         print_account_usage();
     }
     const data = _.merge({}, source, target);
-    data.path = source.path;
+    // TODO: ask Naveen why this line was added
+    //data.path = source.path;
```
Comment (romayalon) [inline src/cmd/manage_nsfs.js:182]: Hey @naveenpaul1 I saw that that source path is always taken, why is that? a user won't be able to update path according to this line

Comment (naveenpaul1) [reply src/cmd/manage_nsfs.js:182]: @romayalon Idea is source(new json schema for update) should be able to update all properties in the existing bucket,  `_.merge()`  was not updating the `path` on its own, So I have to do it in code explicitly.

Comment (romayalon) [reply src/cmd/manage_nsfs.js:182]: but this line - data.path = source.path; won't update the data with a new path, it takes the source and not the new path. or am I missing something?
and thanks i'll update the print

Comment (naveenpaul1) [reply src/cmd/manage_nsfs.js:182]: @romayalon got your point and you are correct, This code takes the same path from the existing bucket schema not using new path. It should be `target.path`.

File: src/cmd/manage_nsfs.js [hunk 2]
```diff
@@ -420,13 +429,9 @@ async function validate_bucket_add_args(data) {
         console.error('Error: User data and bucket path should not be empty');
         return false;
     }
-    if (!data.path.includes(data.name.unwrap())) {
```
Comment (naveenpaul1) [inline src/cmd/manage_nsfs.js:423]: why this check removed?

**Review by naveenpaul1 — APPROVED** [review]
LGTM

File: src/cmd/manage_nsfs.js (hunk 2 above)
Comment (romayalon) [reply src/cmd/manage_nsfs.js:423]: there is no suck restriction - to have the bucket name in the path of the exported directory

**Pull request merged**